app/src/main/java/com/github/sdp_begreen/begreen/adapters/ExpandableListAdapter.kt

```diff
@@ -0,0 +1,110 @@
+package com.github.sdp_begreen.begreen.adapters
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseExpandableListAdapter
+import android.widget.ImageView
+import android.widget.TextView
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.models.Group
+import com.github.sdp_begreen.begreen.models.Item
+
+// This adapter is responsible for displaying expandable lists with groups and their respective items.
+class ExpandableListAdapter(
+    private val context: Context,
+    private val groups: List<Group>
+) : BaseExpandableListAdapter() {
+
+    // Returns the number of groups in the list.
+    override fun getGroupCount(): Int {
+        return groups.size
+    }
+
+    // Returns the number of children (items) in a particular group.
+    override fun getChildrenCount(groupPosition: Int): Int {
+        return groups[groupPosition].items.size
```

Comment:
You should check that the given parameter `groupPosition` is not an _IndexOutOfBound_. Otherwise throw an `IllegalArgumentException`. You can copy paste this check :

if (groupPosition < 0 || groupPosition >= groups.size) {
      throw new IllegalArgumentException()
}

Reply:
Same for other methods

Reply:
Thanks, Ilias, well spotted. I've made the requested changes.